modules/drivers/sensors/hx711/hx711.ts

```diff
@@ -0,0 +1,76 @@
+// Avia HX711 24-bit ADC for weight scales
+// Copyright © 2023 by Thorsten von Eicken.
+
+import Time from "time"
+import HX711c from "embedded:sensor/ADC/HX711c"
+
+export interface Options {
+  sensor: {
+    clk: number
+    din: number
+  }
+  onError?: (error: string) => void
+
+  // gain selects the gain of the ADC as well as the analog input channel
+  gain?: number // 1:128chA, 2:32chB, 3:64chA (default: 1)
+}
+
+export default class HX711 {
+  #hx711c: HX711c
+
+  constructor(options) {
+    this.configure(options)
+  }
+
+  close() {
+    this.#hx711c = undefined
+  }
+
+  // To "configure" the HX711 we have to perform a read, which ends up setting up the gain
+  // for the next read
+  configure(options: Options) {
+    this.#hx711c = new HX711c(options.sensor.clk, options.sensor.din, options.gain || 1)
+    this.#hx711c.read()
+  }
+
+  get format() {
+    return "number"
+  }
+
+  readable() {
+    //return this.#din.read() == 0
+  }
```

Comment:
Is this function necessary? It isn't in the standard. The usual behavior is to have `sample()` return `undefined` if no reading is available.